nvtabular/ops/categorify.py

```diff
@@ -207,10 +215,40 @@ def __init__(
         num_buckets=None,
         vocabs=None,
         max_size=0,
-        start_index=0,
         single_table=False,
         cardinality_memory_limit=None,
+        tree_width=None,
+        split_out=1,
+        split_every=8,
+        **kwargs,  # Deprecated/unsupported arguments
     ):
+        # Handle deprecations and unsupported kwargs
+        if "start_index" in kwargs:
+            raise ValueError(
+                "start_index is now deprecated. `Categorify` will always "
+                "reserve index `0` for user-specific purposes, and will "
+                "use index `1` for null values."
+            )
+        if "na_sentinel" in kwargs:
+            raise ValueError(
+                "na_sentinel is now deprecated. `Categorify` will always "
+                "reserve index `1` for null values, and the following "
+                "`num_buckets` indices for out-of-vocabulary values "
+                "(or just index `2` if `num_buckets is None`)."
+            )
+        if kwargs:
+            raise ValueError(f"Unrecognized key-word arguments: {kwargs}")
+
+        # Warn user if they set num_buckets without setting max_size or
+        # freq_threshold - This setting used to hash everything, but will
+        # now just use multiple indices for OOV encodings at transform time
+        if num_buckets and not (max_size or freq_threshold):
+            warnings.warn(
+                "You are setting num_buckets without using max_size or "
+                "freq_threshold to restrict the number of distinct "
+                "categories. Are you sure this is what you want?"
+            )
```

Comment:
@rnyak - This is one detail I am still unsure about. I really don't want to make it easy for users to "hash everything" (what currently happens when the user **only** sets `num_buckets`).  However, it is still strange for the user to ***only** set `num_buckets` (now for a slightly-different reason).  My issue with the original behavior is that there is really no reason for `Catogorify` to be used at all for the case of "hash everything," because we are doing a lot of unnecessary work that can be accomplished by a simple lambda op.

Reply:
@rjzamora I am ok to take `num_buckets` out of Categorify. So far, we dont know a Merlin user who only wants to use `num_buckets` (IIRC). Not sure anyone wants to really hash entire item catalog only with num_buckets.. then how they will reverse back to original ids is a question mark. but I leave the decision to you and @karlhigley.

Reply:
>I am ok to take num_buckets out of Categorify

Just to clarify, I'm not really proposing that we remove the `num_buckets` option, but that we slightly change it's meaning to **only** do what it used to do when `max_size` or `freq_threshold` was set:

- **OLD meaning**: We hash everything, **UNLESS** `max_size` or `freq_threshold` is also set (in which case we use `num_buckets` to be the number of hash-buckets to use for infrequent OOV values at transformation time).
-  **NEW meaning**: We **always** take `num_buckets` to be the number of hash-buckets to use for infrequent OOV values at transformation time. Even if neither `max_size` nor `freq_threshold` is set, we will still use a hash+modulo for OOV values at transformation time if `num_buckets>1`.